### PR TITLE
chore: replace classnames with clsx

### PR DESCRIPTION
**Pull request by wojtekmaj**
## Description

`clsx` is a drop-in replacement for `classnames`, a tad (~200B) smaller, and a bit more popular (overtook classnames 1.5 yr ago). A no brainer, really.

## Validation

The only errors could be me forgetting to change some `classnames` imports to `clsx` (which would have triggered a lint error), or forgetting to update some `classNames()` calls to `clsx()` (which would have triggered a lint error). So as long as ESLint is happy, I am too!

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npm run format` to ensure the code follows the style guide.
- [x] I have run `npm run test` to check if all tests are passing.
- [ ] I have run `npx turbo build` to check if the website builds without errors.
- [ ] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Comment by avivkeller** [comment]
I honestly don't see a difference between the packages, 200B during build really isn't too much imo, but we'll see what other members say

File: apps/site/package.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "@vercel/otel": "~1.10.0",
     "@vercel/speed-insights": "~1.1.0",
     "autoprefixer": "~10.4.20",
-    "classnames": "~2.5.1",
+    "clsx": "^2.1.1",
```
Comment (canerakdas) [inline apps/site/package.json:64]:
```suggestion
    "clsx": "~2.1.1",
```

IMO we should keep `~`; https://github.com/nodejs/nodejs.org/blob/main/DEPENDENCY_PINNING.md?plain=1#L23

**Review by canerakdas — COMMENTED** [review]
+1 to @RedYetiDev , I think I am in favor of using whatever is best for newcomers

Also, It would be nice to add why we use this utility package to [COLLABORATOR_GUIDE](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md)

**Comment by bmuenzenmeyer** [comment]
I'm ambivalent about this change unless there is appreciable:

- difference in license: both `MIT`
- API benefit - identical enough
- size difference - negligible
- security - neither have a policy in place

**Comment by avivkeller** [comment]
I feel like if I personally had a choice, I would choose `classnames`, as:
1. It's releases are certified by Github Actions (more secure)
2. It's relied on by 4x the dependencies (44,000 vs 11,000)
3. It has double the stars (17k vs 8k)

but those arguments aren't very persuasive, as 8k stars is still a lot, ditto for deps.

**Comment by ovflowd** [comment]
> I feel like if I personally had a choice, I would choose `classnames`, as:
> 
> 1. It's releases are certified by Github Actions (more secure)
> 2. It's relied on by 4x the dependencies (44,000 vs 11,000)
> 3. It has double the stars (17k vs 8k)
> 
> but those arguments aren't very persuasive, as 8k stars is still a lot, ditto for deps.

You mean classnames is used by more people?

**Comment by avivkeller** [comment]
> You mean classnames is used by more people?

Yes.

**Comment by bmuenzenmeyer** [comment]
i am proposing we close this as welcome contribution but not not really a step-change from the current approach.

**Pull request closed without merging**